src/bin/pg_autoctl/cli_drop_node.c

```diff
@@ -578,9 +578,23 @@ cli_drop_local_node(KeeperConfig *config, bool dropAndDestroy)
 	 */
 	pid_t pid = 0;
 
-	if (!wait_for_pid_to_exit(config->pathnames.pid, 30, &pid))
+	/*
+	 * If --force isn't used then any running pg_autoctl process will detect
+	 * that it is dropped and clean itself up nicely and finally it will exit.
+	 * We give the process 30 seconds to exit by itself, if it didn't probably
+	 * something is wrong and we manually kill it. We will then complete the
+	 * transition to dropped from this process instead.
+	 *
+	 * If --force is used, we skip the transition to "dropped". So a currently
+	 * running process won't realise it's dropped, so it will not exit by
+	 * itself. So there's no point in waiting for 30 seconds. Instead we
+	 * manually kill it using SIGQUIT right away.
```

Comment:
Can we bypass calling `wait_for_pid_to_exit` entirely in this case, then?

Reply:
I think it's still nice to stop a currently running pg_autoctl process in this case. The first `wait_for_pid_to_exit` then simply becomes an easy way to check if there is a running process.